symfony/plugins/orangehrmPerformancePlugin/Api/PerformanceTrackerAPI.php

```diff
@@ -0,0 +1,224 @@
+            new ParameterBag([CommonParams::PARAMETER_TOTAL=> $count])
+        );
+    }
+
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    CommonParams::PARAMETER_EMP_NUMBER,
+                    new Rule(Rules::POSITIVE)
+                ),
+            ),
+            ...$this->getSortingAndPaginationParamsRules(PerformanceTrackerSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    public function create(): EndpointResult
+    {
+       $performanceTracker = new PerformanceTracker();
+       $reviewers =$this->getRequestParams()->getArray(
+           RequestParams::PARAM_TYPE_BODY,
+           self::PARAMETER_REVIEWERS
+       );
+       $this->setPerformanceTrackerParams($performanceTracker,1);
+       $this->getPerformanceTrackerService()->getPerformanceTrackerDao()->savePerformanceTracker($performanceTracker,$reviewers);
+       return new EndpointResourceResult(PerformanceTrackerModel::class, $performanceTracker);
+    }
+
+    private function setPerformanceTrackerParams(PerformanceTracker $performanceTracker ,int $num): void
+    {
+        $performanceTracker->setTrackerName(
+            $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_TRACKER_NAME)
+        );
+        $empNumber =$this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_BODY, CommonParams::PARAMETER_EMP_NUMBER);
+        $performanceTracker->getDecorator()->setEmployeeByEmpNumber($empNumber);
+        $addedEmp =$this->getAuthUser()->getEmpNumber();
+        $performanceTracker->getDecorator()->setAddedByByEmpNumber($addedEmp);
+        if($num == 1){
+            $performanceTracker->setAddedDate($this->getDateTimeHelper()->getNow());
+        }else{
+            $performanceTracker->setModifiedDate($this->getDateTimeHelper()->getNow());
+        }
+
+    }
+
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            ...$this->getCommonBodyParamRulesCollection()
+        );
+
+    }
+
+    /**
+     * @return array
+     */
+    protected function getCommonBodyParamRulesCollection(): array
+    {
+        return [
+            new ParamRule(
+                self::PARAMETER_TRACKER_NAME,
+                new Rule(Rules::LENGTH, [null, self::PARAM_RULE_TRACKER_NAME_MAX_LENGTH])
+            ),
+            new ParamRule(
+                CommonParams::PARAMETER_EMP_NUMBER,
+                new Rule(Rules::POSITIVE)
+            ),
+            new ParamRule(
+                self::PARAMETER_REVIEWERS,
+                new Rule(Rules::ARRAY_TYPE)
+            ),
+        ];
+    }
+
+    public function delete(): EndpointResult
+    {
+        $ids = $this->getRequestParams()->getArray(RequestParams::PARAM_TYPE_BODY, CommonParams::PARAMETER_IDS);
```

Comment:
```suggestion
        $ids = $this->getRequestParams()->getArray(
          RequestParams::PARAM_TYPE_BODY, 
          CommonParams::PARAMETER_IDS
        );
```
this will be more readable and clear

Reply:
changed